Student/Student/Courses/Syllabus/SyllabusActionableItemsPresenter.swift

```diff
@@ -0,0 +1,137 @@
+            self?.update()
+        }
+    }()
+
+    private lazy var assignments: Store<GetAssignments> = {
+        let useCase = GetAssignments(courseID: self.courseID, sort: sort)
+        return self.env.subscribe(useCase) { [weak self] in
+            self?.updateAssignments()
+        }
+    }()
+
+    private lazy var calendarEvents: Store<GetCalendarEvents> = {
+        let useCase = GetCalendarEvents(context: ContextModel(.course, id: courseID))
+        return self.env.subscribe(useCase) { [weak self] in
+            self?.updateCalendarEvents()
+        }
+    }()
+
+    init(env: AppEnvironment = .shared, view: SyllabusActionableItemsViewProtocol, courseID: String, sort: GetAssignments.Sort = .position) {
+        self.courseID = courseID
+        self.env = env
+        self.view = view
+        self.sort = sort
+    }
+
+    func viewIsReady() {
+        assignments.exhaust(while: { _ in true })
+        calendarEvents.exhaust(while: { _ in true })
+        course.refresh()
+        update()
+    }
+
+    func update() {
+        var models = assignmentViewModels + calendarViewModels
+        models.sort { (a, b) -> Bool in
+            return (a.dueDate ?? .distantFuture).compare(b.dueDate ?? .distantFuture) == .orderedAscending
+        }
+        view?.update(models: models)
+        loadColor()
+    }
+
+    func updateAssignments() {
+        assignmentUpdateCount += 1
+        assignmentViewModels = assignments.map {
+            SyllabusActionableItemsViewController.ViewModel(id: $0.id, htmlUrl: $0.htmlURL, title: $0.name, dueDate: $0.dueAt, formattedDate: formattedDueDate($0.dueAt), image: icon(for: $0))
+        }
+        signalUpdate()
+    }
+
+    func updateCalendarEvents() {
+        calendarEventUpdateCount += 1
+        calendarViewModels = calendarEvents.map {
+            SyllabusActionableItemsViewController.ViewModel(id: $0.id, htmlUrl: $0.routingURL,
+                                                           title: $0.title, dueDate: $0.endAt, formattedDate: formattedDueDate($0.endAt), image: .icon(.calendarMonth, .line))
+        }
+        signalUpdate()
+    }
+
+    func signalUpdate() {
+        let updateThreshold = 2
+        if(assignmentUpdateCount >= updateThreshold && calendarEventUpdateCount >= updateThreshold) {
+            update()
+        }
+    }
+
+    func loadColor() {
+        guard let course = course.first else { return }
+        view?.updateNavBar(subtitle: course.name, color: course.color)
+    }
+
+    func select(_ htmlURL: URL, from: UIViewController) {
+        env.router.route(to: htmlURL, from: from, options: nil)
+    }
+
+    func formattedDueDate(_ date: Date?) -> String {
+        var result = NSLocalizedString("No Due Date", comment: "")
+        if let date = date {
+            result = DateFormatter.localizedString(from: date, dateStyle: .medium, timeStyle: .short)
+        }
+        return result
+    }
+
+    func icon(for: Assignment) -> UIImage? {
```

Comment:
nitipick: name this function `icon(for assignment: Assignment)` then you don't need "`for`" backtick thingy.

Reply:
Also, I can see this being a helper on `Assignment` so that we can re-use it. We can move it there later though.